stdlib/public/core/StringComparable.swift

```diff
@@ -211,22 +77,18 @@ extension _StringGuts {
   ) -> Int {
     defer { _fixLifetime(left) }
     defer { _fixLifetime(right) }
-#if _runtime(_ObjC)
-    // We only want to perform this optimization on objc runtimes. Elsewhere,
-    // we will make it follow the unicode collation algorithm even for ASCII.
-    // This is consistent with Foundation, but incorrect as defined by Unicode.
-    //
-    // FIXME: String ordering should be consistent across all platforms.
+    
     if left.isASCII && right.isASCII {
       let leftASCII = left._unmanagedASCIIView[leftRange]
       let rightASCII = right._unmanagedASCIIView[rightRange]
       let result = leftASCII.compareASCII(to: rightASCII)
       return result
     }
-#endif
-    return _compareDeterministicUnicodeCollation(
-      _leftUnsafeStringGutsBitPattern: left.rawBits, leftRange,
-      _rightUnsafeStringGutsBitPattern: right.rawBits, rightRange)
+    
+    let selfBits = _explodeBits(left)
```

Comment:
This already exists. See `_StringGuts.rawBits` and `_StringGuts.init(rawBits:)`, and use those for consistency.